docusaurus/docs/engineering/knowledge-base/development-lifecycle.md

```diff
@@ -4,154 +4,230 @@ sidebar_position: 7
 
 # Development Lifecycle
 
-To look at/download BobaBoard's code, check out our guide to [Setting Up your Coding Environment](/docs/engineering/start-developing/before-you-start).
-You might also be interested in [Git Notes](/docs/engineering/legacy-pages/git-notes)
+If you're new to collaborating on a code project, you may be wondering what that process looks like. This page gives an overview of the development lifecycle, explains key terminology, and guides you through the step-by-step process of contributing to BobaBoard's codebase.
 
-:::tip This seems scarier than it is!
-If you're new to coding and github, the following process might seem scary and overwhelming. Don't worry: it will become second nature very quickly! If you want someone to hold your hand throughout the process, contact Ms. Boba.
-:::
+To slowly get you familiar with more technical vocabulary you might run into, this document also includes relevant information about git and GitHub. For more information on these subjects you can look at our [Git Notes](/docs/engineering/legacy-pages/git-notes.md) page.
 
 :::note
-TODO[Ms.Boba]: Explore whether the new github command line interface can make this process less cumbersome.
+We will use GitHub as our example platform since BobaBoard's code is hosted there. However, there are alternative platforms that serve similar purposes.
 :::
 
-## Development lifecycle overview
+As a side note, if you've looked at our [Recommended Tools](/docs/engineering/knowledge-base/recommended-tools.md) document, you might be aware that we recommend Visual Studio Code as a code editor. One reason for this is its excellent integration with git. This lets you perform various tasks using the editor's user interface instead of typing commands in your terminal. Feel free to use integrations if you prefer. However, we will still list the commands here so that everyone can follow along and understand what's happening behind the scenes.
```

Comment:
Unsure how I feel about putting two callouts one after the other, but I would not put this as part of the main document, and would reserve it for a note.

What do you say about moving it to a note near the beginning of the Guide section?